File: 0717-1-bit-and-2-bit-characters/0717-1-bit-and-2-bit-characters.js
/**
 * @param {number[]} bits
 * @return {boolean}
 */
var isOneBitCharacter = function(bits) {
    const n = bits.length;
    let i = 0;
    
    while(i < n - 1){
        i += bits[i] + 1
    }
    
    return i === n - 1;
};

File: 0717-1-bit-and-2-bit-characters/0717-1-bit-and-2-bit-characters.go
func isOneBitCharacter(bits []int) bool {
    i := 0

    for i < len(bits) - 1 { 
        if bits[i] == 0 {
            i += 1
        } else {
            i += 2
        }
    }

    return i == len(bits)-1
}